Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -321,11 +332,23 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
                 self.tunnelHealth.isHavingConnectivityIssues = false
                 self.updateBandwidthAnalyzerAndRekeyIfExpired()
 
-            case .reconnected:
+            case .reconnected(let failureCount):
+                providerEvents.fire(.connectionTesterStatusChange(.recovered(duration: .immediate, failureCount: failureCount)))
+
+                if failureCount >= 8 {
+                    providerEvents.fire(.connectionTesterStatusChange(.recovered(duration: .extended, failureCount: failureCount)))
```

Comment:
Being able to see if users are seeing failures for more than a couple of minutes is good because we want to see if there's a need to offer some form of manual recovery for this case.  This is information that can give us something actionable.